### PR TITLE
py-pycuda: new version 2020.1

### PR DESCRIPTION
**Pull request by aweits**
fix dependencies/depend on spack's boost.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pycuda/package.py [hunk 1]
```diff
@@ -14,13 +14,28 @@ class PyPycuda(PythonPackage):
...
+        boostlib = 'boost_python{0}'.format(pyver)
+        configure_args = [
+            '--no-use-shipped-boost',
+            '--boost-inc-dir={0}'.format(self.spec['boost'].prefix.include),
+            '--boost-lib-dir={0}'.format(self.spec['boost'].prefix.lib),
+            '--boost-python-libname={0}'.format(boostlib)
+        ]
+        python('configure.py', *configure_args)
+
     depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pycuda/package.py:33]: Can you add a dependency on `python@3.6:3.999`? Also need a dep on `py-pytools@2011.2:`

File: var/spack/repos/builtin/packages/py-pycuda/package.py [hunk 2]
```diff
@@ -14,13 +14,28 @@ class PyPycuda(PythonPackage):
...
     depends_on('cuda')
-    depends_on('boost')
+    depends_on('boost+python')
+    depends_on('python@3.6:', type=('build', 'run'), when='@2020.1:')
     depends_on('py-numpy@1.6:', type=('build', 'run'))
+    depends_on('py-pytools@2011.2:', type=('build', 'run'))
     depends_on('py-six', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pycuda/package.py:39]: Six is no longer needed, other deps should be type=build/run

**Review by alalazo — DISMISSED** [review]

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pycuda/package.py (hunk 2 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-pycuda/package.py:39]: six is still needed at runtime:
```shell
[aweits@localhost pycuda]$ grep -r "import six" *
examples/download-examples-from-wiki.py:    import six.moves.xmlrpc_client
pycuda/curandom.py:import six
pycuda/driver.py:import six
pycuda/elementwise.py:import six
pycuda/gpuarray.py:import six
pycuda/gpuarray.py:import six.moves.copyreg
pycuda/scan.py:import six
pycuda/tools.py:import six

```

File: var/spack/repos/builtin/packages/py-pycuda/package.py [hunk 3]
```diff
@@ -14,16 +14,31 @@ class PyPycuda(PythonPackage):
...
     version('2016.1.2', sha256='a7dbdac7e2f0c0d2ad98f5f281d5a9d29d6673b3c20210e261b96e9a2d0b6e37')
 
+    @run_before('build')
+    def configure(self):
+        pyver = self.spec['python'].version.up_to(2).joined
+        boostlib = 'boost_python{0}'.format(pyver)
+        configure_args = [
+            '--no-use-shipped-boost',
+            '--boost-inc-dir={0}'.format(self.spec['boost'].prefix.include),
+            '--boost-lib-dir={0}'.format(self.spec['boost'].prefix.lib),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pycuda/package.py]:
```suggestion
            '--boost-lib-dir={0}'.format(self.spec['boost'].libs.directories[0]),
```
May be in lib or lib64.

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**